### ISSUE TITLE
Set word as the default text in the prompt for `grep_string` 

### ISSUE THREAD

**Issue by krshrimali** [nvim-telescope/telescope.nvim#2822]
**Is your feature request related to a problem? Please describe.**

Currently, on running `require('telescope.builtin').grep_string({})` - the word under the cursor won't appear as the prompt text. Following is the use-case when it would be nice to have:

- User is searching for a word "entry"
- User decides to change something in the given word, quite common when searching for a test class. Example: searching for `GrepStringTest' word, user might decide to just remove the `Test` suffix to see if there is any `GrepString` word.

Currently, the way to do this (please correct me if I'm wrong) is to select the word, paste in the prompt again and then remove the `Test` suffix (/type it themselves).

**Describe the solution you'd like**

- Having the word being searched appear in the prompt by default as the text.

**Describe alternatives you've considered**

- I would generally copy the word and paste in the prompt manually and then attempt the follow-up searches.

**Additional context**

I'm not sure if there is any other way to do this, and it's possible I might have missed a trick/intention here. Please don't hesitate to correct me here, I'm open to opinions and thoughts here. 🙇🏻

**Comment by krshrimali** [comment]
https://github.com/nvim-telescope/telescope.nvim/pull/2823 - Here is the sample PR that I believe could fix this / add this feature. Again, open to discussions on this, and happy to close it if it's not in the line of thoughts of maintainers/contributors. Thank you!

**Comment by jamestrew** [comment]
I'm not sure this is a good idea if I'm understanding your intentions correctly. The idea behind `grep_string` is to allow for fuzzy searching of results from grep (rg) of a given word.

The search results from rg is already "locked in" when the picker opens. So if you search for `GrepStringTest`, even with you PR, having `GrepStringTest` in the prompt, altering the prompt, deleting the `Test`, to `GrepString` won't really help narrow or expand the search results (all instances of `GrepStringTest` already contain the substring `GrepString` and other instances of just the word `GrepString` will not be re-populated).

I think something like this is better suited with `live_grep` like so:
`require("telescope.builtin").live_grep({ default_text=vim.fn.expand("<cword>") })`

What do you think?

**Comment by krshrimali** [comment]
Hi @jamestrew - Thank you so much for taking a look!

You are actually spot-on! 🎉 🚀 I just checked and `grep_string` wasn't really updating the results if I'm trying to modify the prompt. The naming also makes sense, as I should have opted for `live_grep` - the command you shared actually solves my problem well, thank you so much!

Closing this issue in favor of the command you shared, I'll also close my PR now.
